src/modules/utils/motordrivercontrol/MotorDriverControl.cpp

```diff
@@ -112,34 +127,86 @@ bool MotorDriverControl::config_module(uint16_t cs)
         chip= TMC2660;
         tmc26x= new TMC26X(std::bind( &MotorDriverControl::sendSPI, this, _1, _2, _3), axis);
 
+    }else if(str == "TMC2208") {
+        chip= TMC2208;
+        tmc22x= new TMC22X(std::bind( &MotorDriverControl::sendUART, this, _1, _2, _3), axis);
+
     }else{
         THEKERNEL->streams->printf("MotorDriverControl %c ERROR: Unknown chip type: %s\n", axis, str.c_str());
         return false;
     }
 
-    // select which SPI channel to use
-    int spi_channel = THEKERNEL->config->value(motor_driver_control_checksum, cs, spi_channel_checksum)->by_default(1)->as_number();
-    int spi_frequency = THEKERNEL->config->value(motor_driver_control_checksum, cs, spi_frequency_checksum)->by_default(1000000)->as_number();
+    if(chip == TMC2208) {
+        //Configure soft UART
+    		
+        //select TX and RX pins
+        sw_uart_tx_pin = new Pin();
+        sw_uart_rx_pin = new Pin();
+        sw_uart_tx_pin->from_string(THEKERNEL->config->value(motor_driver_control_checksum, cs, sw_uart_tx_pin_checksum)->by_default("nc")->as_string())->as_output();
+        sw_uart_rx_pin->from_string(THEKERNEL->config->value(motor_driver_control_checksum, cs, sw_uart_rx_pin_checksum)->by_default("nc")->as_string())->as_input();
+
+        if(!sw_uart_tx_pin->connected()) {
+            THEKERNEL->streams->printf("MotorDriverControl %c ERROR: uart tx pin not defined\n", axis);
+            return false; 
+        }
+
+        if(sw_uart_rx_pin->connected()) {
+            // Read/Write mode
+            PinName txd = port_pin((PortName)(sw_uart_tx_pin->port_number<<8), sw_uart_tx_pin->pin);
+            PinName rxd = port_pin((PortName)(sw_uart_rx_pin->port_number<<8), sw_uart_rx_pin->pin);
+
+            write_only   = false;
+            this->serial = new BufferedSoftSerial(txd, rxd);
+        } else {
+            // Write only mode
+            PinName txd = port_pin((PortName)(sw_uart_tx_pin->port_number<<8), sw_uart_tx_pin->pin);
```

Comment:
```suggestion
            PinName txd = port_pin((PortName)(sw_uart_tx_pin->port_number), sw_uart_tx_pin->pin);
```

As long as Port is 0 your code is fine. I am trying to run it on a MKS SGEN_L where they assigned Port 1 to the rx/tx pins. 

Note to others:  Port1 does not allow interrupts, so the RX functionallity does not work. No problem though, just use NC as port/pin name in the config